From b60f496005c1a0058a82fb977d9335779add5578 Mon Sep 17 00:00:00 2001
From: "Doug.Smith" <doug.smith@lairdconnect.com>
Date: Fri, 21 Oct 2022 11:07:49 -0400
Subject: [PATCH] fipsinstall-verify:  ensure fips provider is loaded

Ensure that the fips provider is loaded for the
fipsinstall -verify operation.
Was not being loaded if the fipsmodule.cnf activate=1
was commented out for programmatic loading of the fips
module.
---
 apps/fipsinstall.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/apps/fipsinstall.c b/apps/fipsinstall.c
index d0efdf7..97054d2 100644
--- a/apps/fipsinstall.c
+++ b/apps/fipsinstall.c
@@ -484,6 +484,9 @@ opthelp:
     }
 
     if (verify) {
+        if (!load_fips_prov_and_run_self_test(prov_name))
+            goto end;
+
         if (!verify_config(in_fname, section_name, module_mac, module_mac_len,
                            install_mac, install_mac_len))
             goto end;
-- 
2.17.1

